src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java

```diff
@@ -161,4 +173,67 @@ public void sanitizeForSaving() {
         this.googleId = SanitizationHelper.sanitizeGoogleId(this.googleId);
     }
 
+    /**
+     * A Builder class for {@link StudentProfileAttributes}.
+     */
+    public static class Builder {
+        private final StudentProfileAttributes profileAttributes = new StudentProfileAttributes();
+
+        public Builder withGoogleId(String googleId) {
+            profileAttributes.googleId = getOriginalOrDefaultIfNull(googleId);
+            return this;
+        }
+
+        public Builder withShortName(String shortName) {
+            profileAttributes.shortName = getOriginalOrDefaultIfNull(SanitizationHelper.sanitizeName(shortName));
+            return this;
+        }
+
+        public Builder withEmail(String email) {
+            profileAttributes.email = getOriginalOrDefaultIfNull(SanitizationHelper.sanitizeEmail(email));
+            return this;
+        }
+
+        public Builder withInstitute(String institute) {
+            profileAttributes.institute = getOriginalOrDefaultIfNull(SanitizationHelper.sanitizeTitle(institute));
+            return this;
+        }
+
+        public Builder withNationality(String nationality) {
+            profileAttributes.nationality = getOriginalOrDefaultIfNull(SanitizationHelper.sanitizeName(nationality));
+            return this;
+        }
+
+        public Builder withGender(String gender) {
+            profileAttributes.gender = isGenderValid(gender) ? gender : "other";
+            return this;
+        }
+
+        public Builder withMoreInfo(String moreInfo) {
+            profileAttributes.moreInfo = getOriginalOrDefaultIfNull(moreInfo);
+            return this;
+        }
+
+        public Builder withPictureKey(String pictureKey) {
+            profileAttributes.pictureKey = getOriginalOrDefaultIfNull(pictureKey);
+            return this;
+        }
+
+        public Builder withModifiedDate(Date modifiedDate) {
+            profileAttributes.modifiedDate = modifiedDate;
+            return this;
+        }
+
+        private String getOriginalOrDefaultIfNull(String value) {
+            return value == null ? StringHelper.EMPTY : value;
```

Comment:
So the behavior is assigning empty string if the passed value is `null`. Does it seem to be correct?

Alternative: assign only if the supplied value is not `null`.
```java
if (value != null) {
    profileAttributes.attribute = value;
}
```
I am not saying that the alternative is the definitely correct one (albeit it is the commonly used pattern in builders), but I'm asking to consider the different choices.